lib/PuppeteerSharp/Payload.cs

```diff
@@ -23,67 +23,66 @@ internal Payload()
         /// </summary>
         /// <value>HTTP method.</value>
         [JsonProperty("method"), JsonConverter(typeof(HttpMethodConverter))]
-        public HttpMethod Method { get; internal set; }
+        public HttpMethod Method { get; set; }
         /// <summary>
         /// Gets or sets the post data.
         /// </summary>
         /// <value>The post data.</value>
         [JsonProperty("postData")]
-        public object PostData { get; internal set; }
+        public object PostData { get; set; }
         /// <summary>
         /// Gets or sets the HTTP headers.
         /// </summary>
         /// <value>HTTP headers.</value>
         [JsonProperty("headers")]
-        public Dictionary<string, object> Headers { get; internal set; }
+        public Dictionary<string, object> Headers { get; set; }
         /// <summary>
         /// Gets or sets the URL.
         /// </summary>
         /// <value>The URL.</value>
         [JsonProperty("url")]
-        public string Url { get; internal set; }
+        public string Url { get; set; }
 
-        [JsonIgnore]
-        internal string Hash
-        {
-            get
-            {
-                var normalizedUrl = Url;
-
-                try
-                {
-                    // Decoding is necessary to normalize URLs.
-                    // The method will throw if the URL is malformed. In this case,
-                    // consider URL to be normalized as-is.
-                    normalizedUrl = HttpUtility.UrlDecode(Url);
-                }
-                catch
-                {
-                }
-
-                var hash = new Payload
-                {
-                    Url = Url,
-                    Method = Method,
-                    PostData = PostData
-                };
-
-                if (!normalizedUrl.StartsWith("data:", StringComparison.Ordinal))
-                {
-                    foreach (var item in Headers.OrderBy(kv => kv.Key))
-                    {
-                        if (item.Key.Equals("Accept", StringComparison.OrdinalIgnoreCase)
-                            || item.Key.Equals("Referer", StringComparison.OrdinalIgnoreCase)
-                            || item.Key.Equals("X-DevTools-Emulate-Network-Conditions-Client-Id", StringComparison.OrdinalIgnoreCase))
-                        {
-                            continue;
-                        }
-                        hash.Headers[item.Key] = item.Value;
-                    }
-                }
-
-                return JsonConvert.SerializeObject(hash);
-            }
+        internal string Hash

```

Comment:
Shouldn't this still have the JsonIgnore attribute?

Reply:
Why would we need it if we are not serializing this object?

Reply:
In that case remove all of the attributes in this class

Reply:
`JsonProperty` it's used for deserialization. e.g. https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp/Messaging/RequestInterceptedResponse.cs#L13